packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -320,6 +320,58 @@ class AndroidSdk {
 
   String? getAvdManagerPath() => getCmdlineToolsPath(globals.platform.isWindows ? 'avdmanager.bat' : 'avdmanager');
 
+  /// From https://developer.android.com/ndk/guides/other_build_systems.
+  static const Map<String, String> _llvmHostDirectoryName = <String, String>{
+    'macos': 'darwin-x86_64',
+    'linux': 'linux-x86_64',
+    'windows': 'windows-x86_64',
+  };
+
+  String? getNdkBinaryPath(String binaryName) {
+    final Directory ndk = directory.childDirectory('ndk');
+    if (!ndk.existsSync()) {
+      return null;
+    }
+    final List<Version> ndkVersions = ndk
+        .listSync()
+        .map((FileSystemEntity entity) {
+          try {
+            return Version.parse(entity.basename);
+          } on Exception {
+            return null;
+          }
+        })
+        .whereType<Version>()
+        .toList()
+      // Use latest NDK first.
```

Comment:
Can we add a test that validates we pick the latest version?

Reply:
Yes, we did in the same commit. See packages/flutter_tools/test/general.shard/android/android_sdk_test.dart